### PR TITLE
fix #282698: undoing timesig change with tremolo causes crash

### PR DESCRIPTION
**Pull request by jthistle**
Resolves: https://musescore.org/en/node/282698

### CODE REVIEW

File: libmscore/undo.h [hunk 1]
```diff
@@ -1317,5 +1317,27 @@ class FretMarker : public UndoCommand {
...
+      int track;
+
+      Chord* oldC1;
+      Chord* oldC2;
+
+      void undo(EditData*) override;
+      void redo(EditData*) override;
+
+   public:
+      MoveTremolo(Score* s, Fraction c1, Fraction c2, Tremolo* tr, int t) : score(s), chord1Tick(c1), chord2Tick(c2), trem(tr), track(t) {}
```
Comment (anatoly-os) [inline libmscore/undo.h]: It is better to extract constructor as a method to .cpp file. Assigning so many variables litters the code in .h file.

**Comment by anatoly-os** [comment]
Is it possible to add an mtest which checks the scenario?

File: libmscore/undo.h (hunk 1 above)
Comment (jthistle) [reply libmscore/undo.h]: I mean, most of the constructors for undo are in the .h file. But sure, I'll move it if you want.

**Comment by jthistle** [comment]
Possibly... maybe a script recorder test?

**Comment by jthistle** [comment]
Test added @anatoly-os

File: libmscore/undo.h (hunk 1 above)
Comment (jthistle) [reply libmscore/undo.h]: Please can I keep it like this? Literally almost every other constructor is in undo.h, a lot with 4+ variables.

**Pull request merged**